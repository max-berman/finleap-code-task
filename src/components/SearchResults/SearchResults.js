import React from 'react'

export default function SearchResults({ cityList = [], setCityList }) {
  function handleOnClick(itemId) {
    setCityList(cityList.filter(({ id }) => itemId !== id))
  }
  function className(temp) {
    return temp > 20 ? 'hot' : temp >= 5 && temp <= 20 ? 'mild' : 'cold'
  }

  return (
    cityList.length > 0 && (
      <section className='cards'>
        {cityList
          .sort((x, y) => +y.temp_max - +x.temp_max)
          .map(({ id, name, temp_max, temp_min }, key) => (
            <dl className={`${className(temp_max)} card`} key={key}>
              <dd className='close' onClick={() => handleOnClick(id)} />
              <dt>{name}</dt>
              <dd>
                <span>min:</span> {temp_max}
              </dd>
              <dd>
                <span>max:</span> {temp_min}
              </dd>
            </dl>
          ))}
      </section>
    )
  )
}
