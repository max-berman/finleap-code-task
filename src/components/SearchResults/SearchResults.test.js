import React from 'react'
import { render, cleanup } from '@testing-library/react'
import SearchResults from './SearchResults'

afterEach(cleanup)

describe('<SearchResults /> spec', () => {
  it('renders the component', () => {
    const container = render(<SearchResults />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
