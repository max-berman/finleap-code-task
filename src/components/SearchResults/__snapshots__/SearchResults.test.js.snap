// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResults /> spec renders the component 1`] = `undefined`;
